File: 338-counting-bits/counting-bits.cpp
class Solution {
public:
    vector<int> countBits(int n) {
        vector<int> ans(1,0);
        if(n >= 1) ans.push_back(1);
        for(int i=2; i<=n; i++){
            if(i % 2 == 0) ans.push_back(ans[i/2]);
            else ans.push_back(ans[i/2]+1);
        }
        return ans;
    }
};

File: 338-counting-bits/counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        ans = []
        for i in range(0, n+1):
            count = 0
            while(i):
                if i & 1:
                    count += 1
                i = i >> 1
            ans.append(count)
        return ans
        
        